src/PackageLagMonitor/SearchServiceClient.cs

```diff
@@ -177,23 +206,44 @@ private List<Instance> GetInstances(Uri endpointUri, int instanceCount, RegionIn
                     var diagUriBuilder = new UriBuilder(endpointUri);
 
                     diagUriBuilder.Scheme = "https";
-                    diagUriBuilder.Port = instancePortMinimum + i;
+                    if (serviceType == ServiceType.LuceneSearch)
+                    {
+                        diagUriBuilder.Port = instancePortMinimum + i;
+                    }
                     diagUriBuilder.Path = "search/diag";
 
                     var queryBaseUriBuilder = new UriBuilder(endpointUri);
 
                     queryBaseUriBuilder.Scheme = "https";
-                    queryBaseUriBuilder.Port = instancePortMinimum + i;
+                    if (serviceType == ServiceType.LuceneSearch)
+                    {
+                        queryBaseUriBuilder.Port = instancePortMinimum + i;
+                    }
                     queryBaseUriBuilder.Path = "search/query";
 
                     return new Instance(
                         ProductionSlot,
                         i,
                         diagUriBuilder.Uri.ToString(),
                         queryBaseUriBuilder.Uri.ToString(),
-                        regionInformation.Region);
+                        regionInformation.Region,
+                        serviceType);
                 })
                 .ToList();
         }
+
+        private SearchDiagnosticResponse ConvertAzureSearchResponse(AzureSearchDiagnosticResponse azureSearchDiagnosticResponse)
+        {
+            var result = new SearchDiagnosticResponse()
```

Comment:
Nit: you can drop the `()` if you use object initializer syntax

```suggestion
            var result = new SearchDiagnosticResponse
```

Same comment for line 240

Reply:
Done